### PR TITLE
feat(breadcrumbs): Format SQL queries in breadcrumbs

### PR DESCRIPTION
**Pull request by malwilley**
This is a POC! Still requires testing to ensure that the formatting doesn't misfire too often on non-sql content.

This is the second attempt since the first caused too many calls to flagr. I've moved the logic to `DetailedEventSerializer` which should only be used in the event details endpoint.

Before:

CleanShot 2023-04-04 at 15 49 46

After:

CleanShot 2023-04-04 at 15 49 09

### CODE REVIEW

File: src/sentry/api/serializers/models/event.py [hunk 1]
```diff
@@ -366,12 +369,36 @@ def _get_perf_problem(self, attrs):
...
+
+        try:
+            for breadcrumb_item in breadcrumbs["data"]["values"]:
+                if breadcrumb_item["category"] in FORMATTED_BREADCRUMB_CATEGORIES:
+                    breadcrumb_item["messageRaw"] = breadcrumb_item["message"]
+                    breadcrumb_item["message"] = sqlparse.format(
+                        breadcrumb_item["message"], reindent_aligned=True
+                    )
+            return event_data
+        except Exception:
```
Comment (barkbarkimashark) [inline src/sentry/api/serializers/models/event.py:392]: Should we log a message here on exception to make it easier to fix the issue?

File: tests/sentry/api/serializers/test_event.py [hunk 2]
```diff
@@ -469,3 +469,28 @@ def test_performance_problem_no_stored_data(self):
...
+            result = serialize(event, None, DetailedEventSerializer())
+
+            assert result["entries"][0]["type"] == "breadcrumbs"
+            # First breadcrumb should not have a message_formatted property
+            assert result["entries"][0]["data"]["values"][0]["message"] == "should not format this"
+            # Second breadcrumb should have whitespace added in message_formatted
+            assert (
+                result["entries"][0]["data"]["values"][1]["message"]
+                == "select *\n  from table\n where something = $1"
+            )
```
Comment (barkbarkimashark) [inline tests/sentry/api/serializers/test_event.py:496]: Small thing: can we assert on `"messageRaw"` value.

File: src/sentry/api/serializers/models/event.py [hunk 3]
```diff
@@ -366,12 +369,36 @@ def _get_perf_problem(self, attrs):
         converted_problem["issueType"] = get_group_type_by_type_id(issue_type).slug
         return converted_problem
 
+    def _format_breadcrumb_messages(
+        self, event_data: dict[str, Any], event: Event | GroupEvent, user: User
+    ):
+        breadcrumbs = next(
+            filter(lambda entry: entry["type"] == "breadcrumbs", event_data["entries"]), None
```
Comment (barkbarkimashark) [inline src/sentry/api/serializers/models/event.py:376]: is `entries` guaranteed to exist in `event_data`? Maybe `event_data.get("entries", ())`?

File: src/sentry/api/serializers/models/event.py (hunk 1 above)
Comment (malwilley) [reply src/sentry/api/serializers/models/event.py]: Good call - this should make a sentry issue now

**Review by barkbarkimashark — APPROVED** [review]

**Pull request merged**